src/compat/posix/time/strptime.c

```diff
@@ -0,0 +1,322 @@
+/**
+ * @file strptime.c
+ * @brief writing linux function strptime for embox os
+ *
+ * author Tamara Deryugina <tamara.deryugina@gmail.com>
+ * @date 02.03.2020
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include <ctype.h>
+#include <math.h>
+
+static int num_length(const char *s) {
+	int count = 0;
+	while ((s[count] >= '0') && (s[count] <= '9')) {
+		count++;
+	}
+	return count;
+}
+
+static int word_length(const char *s) {
+	int count = 0;
+	while ((s[count] >= 'A') && (s[count] <= 'z')) {
+		count++;
+	}
+	return count;
+}
+
+static int cmp_before_percent(const char *s, const char *format) {
+	int count = 0;
+	while (format[count] != '\0') {
+		if (format[count] == '%') {
+			return count;
+		}
+		if (strncmp(&format[count], &s[count], 1) != 0) {
+			return 0;
+		}
+		count++;
+	}
+	return count;
+}
+
+static int cmp_symb(const char *s, const char *format, struct tm *tm, int w_leng) {
+	int count = 2;
+	int month = 0;
+	int wday = 0;
+	char arr_day[7][10] = {"sunday", "monday", "tuesday", "wednesday", "thursday", "friday", "saturday"};
+	char arr_month[13][10] = {"january", "february", "march", "april", "may", "june",
+		"july", "august", "september", "october", "november", "december"};
+
+	if ((format[0] == 'a') || (format[0] == 'A')) {
+		while (wday != 7) {
+			while (count != w_leng) {
+				if (tolower(s[count]) != arr_day[wday][count]) {
+					wday++;
+				} else {
+					count++;
+				}
+			}
+		}
+		if (wday == 7) {
+			return 0;
+		}
+		tm->tm_wday = wday;
+		return count;
+
+	} else if ((format[0] == 'b') || (format[0] == 'B') || (format[0] == 'h')) {
+		while (month != 12) {
+			while (count != w_leng) {
+				if (tolower(s[count]) == arr_month[month][count]) {
+					count++;
+				} else {
+					month++;
```

Comment:
Should `count` variable be reset here too?

Reply:
Yes, you're right.
I fixed it.
Thanks!